src/contract.rs

```diff
@@ -254,13 +311,36 @@ pub fn execute_mint_path(
     path: String,
 ) -> Result<Response, ContractError> {
     let config = CONFIG.load(deps.storage)?;
-    let payment_details = PAYMENT_DETAILS.may_load(deps.storage)?;
+    let mut payment_details = PAYMENT_DETAILS.may_load(deps.storage)?;
     if config.token_id.is_none() {
         // No token to mint off of
         return Err(ContractError::NoRootToken {});
     }
     let token_id = config.token_id.unwrap();
 
+    let is_in_claim_window_ = is_in_claim_window(
+        config.reserve_root_for_n_blocks,
+        config.initial_height,
+        env.block.height,
+    );
+
+    if is_in_claim_window_ {
+        if !config.reserve_root_names {
```

Comment:
nit: I think this would be more idiomatic `ensure_eq!(config.reserve_root_names, true, ContractError::ReserveRootNameDisabled {})`